blog: Add "Dev snapshot: Godot 4.1 dev 1" (#626)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-1-dev-1.md

---
title: "Dev snapshot: Godot 4.1 dev 1"
excerpt: "Here be dragons! First development snapshot for the upcoming Godot Engine 4.1 is available to early adopters and adventurous types."
categories: ["pre-release"]
author: Yuri Sizov
image: /storage/blog/covers/dev-snapshot-godot-4-1-dev-1.jpg
image_caption_title: Aether's Edge
image_caption_description: A game by Ancient Stone Studios
date: 2023-04-21 16:00:00
---

> As we have [announced earlier this week](/article/release-management-4-1/) Godot development is ramping up and you can expect Godot Engine 4.1 to be stable before mid-Summer this year. Right now we are in the middle of the feature merging phase, with the main focus being on improving usability, performance, and the feature set of the engine based on [your feedback](https://github.com/godotengine/godot/issues) and [our prioritized plans](/article/rendering-priorities-4-1/).

Reviewer: I'm a bit concerned that the last sentence makes it sound like the focus for everyone is on rendering features. 

It might be better to just say:
```suggestion
As we have [announced earlier this week](/article/release-management-4-1/) Godot development is ramping up and you can expect Godot Engine 4.1 to be stable before mid-Summer this year. Right now we are in the middle of the feature merging phase, with the main focus being on improving usability, performance, and stability.
```

Author: We sometimes link to our previous posts in that manner. Give a bit of credit to our readers ;)

> Many of you have told us how they enjoyed regular preview builds of the engine during the later stages of the Godot 4.0 development. And we have enjoyed getting quick response, reports of new issues and regressions, and opinions about recently implemented features. So we are going to continue this practice going forward, and to start things of today we're releasing the first dev snapshot of Godot 4.1.

Reviewer: ```suggestion
Many of you have told us how you enjoyed regular preview builds of the engine during the later stages of the Godot 4.0 development. And we have enjoyed getting quick responses, reports of new issues and regressions, and opinions about recently implemented features. So we are going to continue this practice going forward, and to start things off today we're releasing the first dev snapshot of Godot 4.1.
```

> Unlike our previous release cycles, 4.1-dev builds are an official part of the process, and you will receive new versions regularly leading up to the beginning of the testing phase and the first beta. Beware that these builds are not guaranteed to be stable or bug-free. Make sure to back up your work or, better yet, to use a version control system. Still, our expectation is that they will be about as stable as Godot 4.0 betas were, so you can use that as your point of reference.

Reviewer: ```suggestion
Starting with today's snapshot, our plan is to regularly release dev snapshots throughout our development period, even before starting an alpha/beta phase. Beware that these builds are not guaranteed to be stable or bug-free and may contain features that have only been lightly tested. Make sure to back up your work or, better yet, use a version control system.
```

- Shaders: Fix completion of `source_color` hint for texture arrays in shaders ([GH-75831](https://github.com/godotengine/godot/pull/75831)).
- XR: Add support for getting and setting display refresh rate in WebXR ([GH-72938](https://github.com/godotengine/godot/pull/72938)).
- XR: Add a get_system_info method to XRInterface ([GH-74848](https://github.com/godotengine/godot/pull/74848)).
- Thirdparty: HarfBuzz 7.1.0, thorvg 0.8.4, mbedtls 2.28.3
- Documentation and translation updates.

This release is built from commit [db1302637](https://github.com/godotengine/godot/commit/db1302637023168f7becceb1c4ce13228e1b2a43).

## Downloads

The downloads for this dev snapshot can be found directly on our repository:

* [Standard build](https://downloads.tuxfamily.org/godotengine/4.1/dev1/) (GDScript, GDExtension).
* [.NET 6 build](https://downloads.tuxfamily.org/godotengine/4.1/dev1/mono) (C#, GDScript, GDExtension).
  - Requires [.NET SDK 6.0](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) or [7.0](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) installed in a standard location.

## Bug reports

> As a tester, you are encouraged to [open bug reports](https://github.com/godotengine/godot/issues) if you experience issues with this release. Please check the [existing issues on GitHub](https://github.com/godotengine/godot/issues) first, using the search function with relevant keywords, to ensure that the bug you experience is not already known.

Reviewer: ```suggestion
As a tester, we encourage you to [open bug reports](https://github.com/godotengine/godot/issues) if you experience issues with this release. Please check the [existing issues on GitHub](https://github.com/godotengine/godot/issues) first, using the search function with relevant keywords, to ensure that the bug you experience is not already known.
```

Author: This is a standard block, FYI. We have it in every post.
